ls1intum/Artemis
# Bugfix: Reference error, ace is not defined

### Checklist

### Motivation and Context
Fixes #2385

### Description
**Bugfix 1:**
There seemed to be an error when trying to access exam as a student. It is related to the dependencies loaded within the exam mode which includes the code-editor dependencies. These are also loaded if there is no programming exercise in the student exam. @schultek suggested that this is because the `code-editor-ace.component` first loads the `brace dependencies` before loading `ace`. The brace dependencies however try to reference `ace` which leads to: `ReferenceError: ace is not defined`. It is interesting that this behavior is not very consistent. Sometimes this error does not show up. 
However: Exam `examPR2415` in https://vmbruegge60.in.tum.de/#/courses/708/exams throws this error consistently.
**Bugfix 2:** 
We had an unhandled case in the client, when the student tries to access the summary but has not submitted on time. It would briefly show the exam end page which states `you have not submitted on time` and then show a blank page because it threw an unhandled error. 

### Steps for Testing
1. Navigate to https://vmbruegge60.in.tum.de/#/courses/708/exams
2. Try to access exam `examPR2415`.
3. Check the console, you should not see this error: 
    `ReferenceError: ace is not defined
    at Object.946 (code-editor-ace.component.ts:1)`
Note: As you are not registered to the exam you cannot participate in it. It should show: "You are not registered to the exam. Contact your instructor" and throw a 404 error.
-------------------------------------
1. Create an exam and set it in the future. 
2. Register student and generate student exams & prepare exercises
3. Change the start date to now.
4. Wait until time runs out and you can't submit anymore.
5. Navigate away from the exam & navigate back to the exam
6. You should still see the note that your exam will not be graded.
`OR`
1. Log in with test user 4 on https://vmbruegge60.in.tum.de
2. navigate to /#/courses/1939/exams/434
3. You should still see the note that your exam will not be graded because you did not submit on time
